### DISCUSSION TITLE
[Questions] The road to v1.0

### DISCUSSION THREAD

**Discussion by Zeitsperre** [Ouranosinc/xclim#1750] [Questions] [answered]
### Setup Information

* Xclim version: v.0.X.X

### Context

We're about to cross into v0.50.0 and have had a milestone open for v1.0.0 for several years. As fun as it is to count ourselves among [0ver](https://0ver.org/) projects like `numba`, `pint`, and `seaborn`, nothing says we need to stay that way.

We adopted true Semantic Versioning not too long ago (though we had something very similar before). [Here's what the SemVer spec has to say about v1.0.0](https://semver.org/spec/v2.0.0.html#how-do-i-know-when-to-release-100).

Not only have we been published under JOSS, but we are also carefully considerate of the breaking changes that are introduced between each minor release.

One metric we considered a while back was to wait to until we have no breaking changes for at least 3 releases and then perform a major release to signify a stable API. Is this still of interest to the developers?

If we can guarantee the stability of our API, that could help get the ball rolling on issues like becoming the backend for [climate_indices](https://github.com/monocongo/climate_indices) (see: #1273).

Opening a discussion here to supplement discussion for the next dev meeting.

**Comment by SarahG-579462** [comment]
I think a necessary step will be to decide whether or not to split off SDBA into it's own project, which will probably require a few versions of warn/deprecate/remove (or we just keep it as a dependency to xclim, but I think that kind of defeats the purpose of splitting it off) ?

**Reply by Zeitsperre** [reply]
I think we are going to keep `SDBA` at the very least as an optional module in perpetuity. It's mentioned in the article and is a major feature for `xclim` users. Your evaluation is right that we would need to provide at least a few versions of warnings for users before we can consider that change to be "stable".

We'll be going more into details about why we want to split this module off at the next meeting of the devs.

**Comment by Zeitsperre** [comment, marked as answer]
Several years later, we have not only split SDBA into https://github.com/Ouranosinc/xsdba, but we are also on the verge of releasing `xclim` v1.0 before the end of year!

Closing this dicussion as resolved.
